### ISSUE TITLE
Segfault in curl_easy_reset (pop_curl_entry returned ce->curl=NULL)

### ISSUE THREAD

**Issue by luke-jr** [luke-jr/bfgminer#46]
After a long (how long?) time with commit 2d6375c354660119d9457ffeeaf14dd0a730373f (2012-08-01)

Very quickly on one startup of bfgminer-2.6.1 (2012-08-05):

```
Core was generated by `./bfgminer-43541e8a -c puppet.json -S icarus:/dev/ttyUSB1 -S icarus:/dev/ttyUSB'.
Program terminated with signal 11, Segmentation fault.
#0  0xf76e1714 in curl_easy_reset () from /usr/lib/libcurl.so.4
(gdb) bt
#0  0xf76e1714 in curl_easy_reset () from /usr/lib/libcurl.so.4
#1  0x0805deb7 in json_rpc_call (curl=0x0, url=0x80efce8 "http://api.bitcoin.cz:8332", 
    userpass=0x80fcc08 "luke-jr.FPGAs:x", 
    rpc_req=0x80d18e4 "{\"method\": \"getwork\", \"params\": [], \"id\":0}\r\n", probe=false, longpoll=false, 
    rolltime=0x81dc4a8, pool=0x80f0888, share=false) at util.c:449
#2  0x0804fbd0 in get_upstream_work (work=0x81dc360, curl=0x0) at miner.c:2111
#3  0x08050526 in get_work_thread (userdata=0xf0900740) at miner.c:2384
#4  0xf7719ed3 in start_thread () from /lib/libpthread.so.0
#5  0xf75823ee in clone () from /lib/libc.so.6
```

Core file: core.keepme_bfgminer-2.6.1_issue46

**Comment by luke-jr** [comment]
Happened again today, got a core dump and backtrace. Somehow a curl entry with NULL curl is getting into the list!

**Comment by luke-jr** [comment]
Possibly fixed with 93ca917 or 1946c80; also added 92c0876 to aide in debugging if it happens again.
